### PR TITLE
[MeshModel] Add workflow and script to generate components | CU-3yvwk9h

### PR DESCRIPTION
**Pull request by humblenginr**
**Description**

This PR 
- Adds a script that generates two files: `component_models.yaml` and `components.yaml` which contain the system metadata and components generated from those systems respectively
- It uses utilities from meshkit to generate components

This script generates **~18500** components in around **15 minutes**. The concurrency parameters such as [this](https://github.com/meshery/meshery/pull/6387/files#diff-76d1b6fe5a5ca5822c004882b83a214e655a9e59311faf7411411d6fc2f04787R108 ) can be modified later after observing for some time to make it even more efficient. 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
Task linked: [CU-3yvwk9h Cronjob for pre-generating components in GitHub Action in meshery-extensions-packages](https://app.clickup.com/t/3yvwk9h)

File: scripts/component_generation/go.mod [hunk 1]
```diff
@@ -0,0 +1,136 @@
+module github.com/sudo-NithishKarthik/meshmodel_ahub_gen
```
Comment (leecalcote) [inline scripts/component_generation/go.mod:1]: We'll need to relocate this package.

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: scripts/component_generation/main.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
+package main
```
Comment (leecalcote) [inline scripts/component_generation/main.go:1]: `/scripts` - where is the logic for self-registering apps and filters located currently?

File: scripts/component_generation/main.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+}
+
+func getApiResponse(offset int, facet bool) (map[string]([]map[string]interface{}), error) {
+	reqUrl := fmt.Sprintf("%s?offset=%v&limit=%s&facets=%v&kind=%v&deprecated=false", AhSearchEndpoint, offset, Limit, facet, HELMKind)
+	resp, err := http.Get(reqUrl)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != 200 {
+		err = fmt.Errorf("status code %d for %s", resp.StatusCode, reqUrl)
```
Comment (leecalcote) [inline scripts/component_generation/main.go:185]: MeshKit Error?

File: .github/workflows/generators.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+name: ArtifactHub Component Generation
```
Comment (leecalcote) [inline .github/workflows/generators.yml:1]: Considering other repos in the future, should this be a generic title?

File: scripts/component_generation/go.mod (hunk 1 above)
Comment (leecalcote) [inline scripts/component_generation/go.mod:1]: Will need to land this module elsewhere.

**Review by leecalcote — CHANGES_REQUESTED** [review]
@Revolyssup please review and confirm that the same approach taken for static component generation in Meshery Adapters is being performed here. There should be no code duplication between static generation and dynamic generation of components (i.e. when a user imports a helm chart containing CRDs). 

Given this, I don't immediately consider that `/scripts` is an appropriate location for core component generation logic.

File: .github/workflows/generators.yml (hunk 4 above)
Comment (sudo-NithishKarthik) [reply .github/workflows/generators.yml]: Changed to `Static Component Generation`

File: scripts/component_generation/go.mod (hunk 1 above)
Comment (sudo-NithishKarthik) [reply scripts/component_generation/go.mod]: moved to `layer5io/meshmodel_ahub_gen`

**Review by leecalcote — APPROVED** [review]

**Pull request merged**